File: src/main/java/com/jinloes/prpilot/util/ProcessUtil.java
package com.jinloes.prpilot.util;

import java.io.File;
import java.util.List;

public final class ProcessUtil {

    private ProcessUtil() {}

    /**
     * Returns the first path in {@code candidates} that points to an existing file, or {@code name}
     * as a fallback for OS PATH resolution.
     */
    public static String findBinary(String name, List<String> candidates) {
        return candidates.stream().filter(p -> new File(p).isFile()).findFirst().orElse(name);
    }
}

File: core/src/jsMain/kotlin/com/jinloes/prpilot/util/ProcessUtil.kt
package com.jinloes.prpilot.util

@JsModule("fs")
@JsNonModule
private external object Fs {
    fun existsSync(path: String): Boolean
    fun statSync(path: String): dynamic
}

actual object ProcessUtil {
    actual fun findBinary(name: String, candidates: List<String>): String =
        candidates.firstOrNull { path ->
            try {
                if (!Fs.existsSync(path)) return@firstOrNull false
                val stat = Fs.statSync(path)
                // stat.isFile() is a function in Node.js
                stat.isFile().unsafeCast<Boolean>()
            } catch (_: Throwable) {
                false
            }
        } ?: name
}
